.github/workflows/compile-test.yml

```diff
@@ -36,30 +37,25 @@ jobs:
             "rocketbootstrap/rocketbootstrap_dynamic.h" # noted to not include directly
             "libundirect/libundirect_hookoverwrite.h" # user's responsibility to include either the base or dynamic header
           )
-          
+
           HEADER_DIR="headers"
           # cut to drop the leading directory
           for HEADER in $(find "${HEADER_DIR}" -name "*.h" | cut -c"$(( ${#HEADER_DIR} + 2))"-); do
-            
+
             for UNCHECKED_DIRECTORY in "${UNCHECKED_DIRECTORIES[@]}"; do
               if [[ "${HEADER}" == "${UNCHECKED_DIRECTORY}"/* ]]; then
                 continue 2
               fi
             done
-            
+
             for UNCHECKED_FILE in "${UNCHECKED_FILES[@]}"; do
               if [[ "${HEADER}" == "${UNCHECKED_FILE}" ]]; then
                 continue 2
               fi
             done
-            
-            {
-              # Eventually we want to not require UIKit, but we're not there yet
-              echo "#import <UIKit/UIKit.h>"
-              echo "#import <${HEADER}>"
-              echo "int main() {}"
-            } | clang -I "${HEADER_DIR}" -isysroot sdks/iPhoneOS14.5.sdk \
-              -target arm64-apple-ios14.5 -arch arm64 \
-              -x "${{ matrix.language }}" ${{ matrix.cflags }} \
-              -fsyntax-only -
+
+            clang -I "${HEADER_DIR}" -isysroot sdks/iPhoneOS14.5.sdk \
+                -target arm64-apple-ios14.5 -arch arm64 \
+                -x "${{ matrix.language }}" ${{ matrix.cflags }} \
+                -stdlib=libc++ -fsyntax-only "${HEADER_DIR}/${HEADER}"
```

Comment:
Why do we need to specify the stdlib?

Reply:
Clang would complain about the lack of stdc++ headers when compiling for ++ variants (see [here](https://github.com/L1ghtmann/headers/actions/runs/6921090617/job/18826290620)). Theos [passes in](https://github.com/theos/theos/blob/7e1f13a85a129620626716148e644461ab1294c6/makefiles/targets/_common/iphone.mk#L17) `-stdlib=libc++`, so this will match current behavior for iOS 12+. 

```clang: warning: include path for libstdc++ headers not found; pass '-stdlib=libc++' on the command line to use the libc++ standard library instead [-Wstdlibcxx-not-found]```

Reply:
Interesting. Do we know if this was happening before this PR, or because this PR includes `includes` of `stdbool.h`/ `stdint.h`?

I think this is ok, but my nit request here would be that we only specify `-stdlib=libc++` when compiling for Objective-C++

Reply:
Seems it was happening before. See [here](https://github.com/leptos-null/theos-headers/actions/runs/6520426342/job/17707970582)

Reply:
Thanks for checking. Sounds good, then only a nit to set the flag for `cpp` only (matches theos as well, I believe)

Reply:
[Done](https://github.com/theos/headers/pull/89/commits/b674600ffa7b8c249fc4534018d219a8ae7916c3)